bindings/c/Cargo.toml

```diff
@@ -25,7 +25,7 @@ homepage = "https://opendal.apache.org/"
 license = "Apache-2.0"
 repository = "https://github.com/apache/opendal"
 rust-version = "1.75"
-version = "0.45.1"
+version = "0.46.0"
```

Comment:
Why do we need to bump this? It seems there are no breaking changes for C and C++.